File: SyllabusManager/SyllabusManager.Data/Models/Syllabuses/SyllabusDescription.cs
using SyllabusManager.Data.Enums.Syllabuses;
using System.ComponentModel.DataAnnotations;

namespace SyllabusManager.Data.Models.Syllabuses
{
    public partial class SyllabusDescription : ModelBase
    {
        public int NumOfSemesters { get; set; }
        public int Ects { get; set; }
        [Required]
        public string Prerequisites { get; set; }
        public ProfessionalTitle ProfessionalTitleAfterGraduation { get; set; }
        [Required]
        public string EmploymentOpportunities { get; set; }
        [Required]
        public string PossibilityOfContinuation { get; set; }
        public GraduationForm FormOfGraduation { get; set; }
    }
}

File: src/frontend/src/app/core/models/syllabus/syllabus-description.ts
import { GraduationForm } from "../../enums/syllabus/graduation-form.enum";
import { ProfessionalTitle } from "../../enums/syllabus/professional-title.enum";
import { ModelBase } from "../model-base";

export class SyllabusDescription extends ModelBase {
    numOfSemesters: number = 0;
    ects: number = 0;
    prerequisites: string = '';
    professionalTitleAfterGraduation: ProfessionalTitle = ProfessionalTitle.BachelorOfScience;
    employmentOpportunities: string = '';
    possibilityOfContinuation: string = '';
    formOfGraduation: GraduationForm = GraduationForm.DiplomaExam;
}
